linux/src/main/java/com/github/weisj/darkmode/platform/linux/gnome/GnomeGtkThemeMonitor.java

```diff
@@ -0,0 +1,172 @@
+package com.github.weisj.darkmode.platform.linux.gnome;
+
+import com.github.weisj.darkmode.platform.ThemeMonitorService;
+import com.github.weisj.darkmode.platform.linux.util.ProcessResult;
+
+import java.io.IOException;
+import java.nio.file.*;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import static com.github.weisj.darkmode.platform.LibraryUtil.userHomeDirectory;
+import static com.github.weisj.darkmode.platform.linux.util.CmdlineUtils.exec;
+import static java.util.Objects.isNull;
+
+/**
+ * This class monitors the GTK theme which is set for the current Gnome environment and makes an estimated guess as to
+ * whether the GTK theme is meant to resemble a "light theme", "dark theme", and whether the theme is "high contrast".
+ * <p>
+ * TODO: Remove the guessing feature in favor of a settings panel where users can specify their themes themselves.
+ */
+public class GnomeGtkThemeMonitor implements Runnable {
```

Comment:
I think the name `GnomeGtkThemeMonitor` is misleading in the context of the other classes. There already is `ThemeMonitor` and this class doesn't exactly as the same notion as it (although it still monitors of course). Maybe `GnomeSettingsWatcher` would be a better name for it. The name of the `run()` method could become `whatchSettings()` for example. Then the constructor of `Thread`  would take `settingsWatcher::whatchSettings`.

Reply:
GnomeGtkThemeMonitor has been replaced with native C++ in 610a6ebc3ab8d8ecb73ba32593a245a2b4502cc8.